nixos-modules/host.nix

```diff
@@ -224,6 +224,9 @@ in
         restartTriggers = [guestConfig.system.build.toplevel];
         overrideStrategy = "asDropin";
       };
+      "microvm-tap-interfaces@${name}" = microvmConfig.restartIfChanged;
+      "microvm-pci-devices@${name}" = microvmConfig.restartIfChanged;
+      "microvm-virtiofsd@${name}" = microvmConfig.restartIfChanged;
```

Comment:
What are you trying to accomplish? Is assigning a boolean to a service definition even valid?
Since these are instance overrides you also need to specify an override strategy like I did above.

I think to fix the root issue you would need to explicitly add `X-RestartIfChanged` to the service or am I missing something?

Reply:
I should also write the inherit and good point with the dropin.

Reply:
There still is the issue that what you are doing currently won't work if `restartIfChanged = false` on the template parent.
See https://github.com/NixOS/nixpkgs/blob/c1bca7fe84c646cfd4ebf3482c0e6317a0b13f22/nixos/lib/systemd-lib.nix#L378-L380
it only sets `X-RestartIsChanged=false` and never `X-RestartIsChanged=true` which causes the actual bug here.